airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/core/context/AirbyteConnectorRunner.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.core.context
+
+import io.airbyte.cdk.core.context.env.ConnectorConfigurationPropertySource
+import io.micronaut.configuration.picocli.MicronautFactory
+import io.micronaut.context.ApplicationContext
+import io.micronaut.context.ApplicationContextBuilder
+import io.micronaut.context.env.CommandLinePropertySource
+import io.micronaut.context.env.Environment
+import io.micronaut.core.cli.CommandLine as MicronautCommandLine
+import picocli.CommandLine
+
+/**
+ * Custom Micronaut application runner that configures the CLI components and adds the custom
+ * property source used to turn the arguments into configuration properties. <p/> Main classes
+ * should use this class as follows: <code>
+ * ```
+ *     public static void main(final String[] args) {
+ *         AirbyteConnectorRunner.run(IntegrationCommand.class, args);
+ *     }
+ * ```
+ * </code>
+ */
+class AirbyteConnectorRunner {
+    companion object {
+        fun <R : Runnable?> run(
+            cls: Class<R>,
+            vararg args: String,
+        ) {
+            val builder: ApplicationContextBuilder = buildApplicationContext(cls, args)
+            builder.start().use { ctx -> run(cls, ctx, *args) }
+        }
+
+        fun <R : Runnable?> run(
+            cls: Class<R>?,
+            ctx: ApplicationContext?,
+            vararg args: String?,
```

Comment:
Why are these three arguments all nullable? We don't want to pass nulls here ever, do we?

Reply:
@postamar Java -> Kotlin generation.  Will fix.